### PR TITLE
e2e tests: Enables should test kubelet managed /etc/hosts file for Windows

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind failing-test
/sig testing
/sig windows

#### What this PR does / why we need it:

We previously skipped this test because Docker did not support mounting individual files in Containers.

Since then, support for Docker has been removed, and containerd on Windows supports this feature. We can run this test on Windows.

#### Which issue(s) this PR fixes:

Related #https://github.com/kubernetes/kubernetes/issues/124426

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/node/kubelet_etc_hosts.go [hunk 1]
```diff
@@ -112,6 +113,12 @@ func assertManagedStatus(
 
 	retryCount := 0
 	etcHostsContent := ""
+	etcHostsPath := etcHostsPathLinux
+	etcHostsOriginalPath := etcHostsOriginalPathLinux
+	if framework.NodeOSDistroIs("windows") {
+		etcHostsPath = etcHostsPathWindows
+		etcHostsOriginalPath = etcHostsOriginalPathWindows
+	}
```
Comment (bart0sh) [inline test/e2e/common/node/kubelet_etc_hosts.go:121]: Looks good to me, just one nit: can we somehow avoid duplicating this in 3 places?

Comment (claudiubelu) [reply test/e2e/common/node/kubelet_etc_hosts.go:121]: Done

**Comment by pacoxu** [comment]
/assign @kubernetes/cncf-conformance-wg  @johnbelamaric @dims 
for conformance-behavior part

**Review by SergeyKanzhelev — APPROVED** [review]
as a general rule we are promoting to Conformance after it was running as non-confirmance for a while. I cannot find the official policy quickly.

In this case changes are quite straightforward, we may want to take an exception here. Test is passing in `/test pull-kubernetes-e2e-capz-windows-master`.

/lgtm
/approve

**Comment by pacoxu** [comment]
/assign @dims
for conformance

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve
/assign @dims

**Pull request merged**